stackit/internal/services/serverupdate/serverupdate_acc_test.go

```diff
@@ -0,0 +1,231 @@
+package serverupdate_test
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/stackitcloud/stackit-sdk-go/core/config"
+	"github.com/stackitcloud/stackit-sdk-go/core/utils"
+	"github.com/stackitcloud/stackit-sdk-go/services/serverupdate"
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/core"
+	"github.com/stackitcloud/terraform-provider-stackit/stackit/internal/testutil"
+)
+
+// Server update schedule resource data
+var serverUpdateScheduleResource = map[string]string{
+	"project_id":           testutil.ProjectId,
+	"server_id":            testutil.ServerId,
+	"update_schedule_name": testutil.ResourceNameWithDateTime("server-update-schedule"),
+	"rrule":                "DTSTART;TZID=Europe/Sofia:20200803T023000 RRULE:FREQ=DAILY;INTERVAL=1",
+	"update_name":          testutil.ResourceNameWithDateTime("server-update-schedule-update"),
+}
+
+func resourceConfig(retentionPeriod int64) string {
+	return fmt.Sprintf(`
+				%s
+
+				resource "stackit_server_update_schedule" "test_schedule" {
+					project_id = "%s"
+					server_id  = "%s"
+					name  = "%s"
+ 				 	rrule = "%s"
+                    enabled = true
+                    update_properties = {
+                        name = "%s"
+                        retention_period = %d
+                        volume_ids = null
+                    }
+				}
+				`,
+		testutil.ServerUpdateProviderConfig(),
+		serverUpdateScheduleResource["project_id"],
+		serverUpdateScheduleResource["server_id"],
+		serverUpdateScheduleResource["update_schedule_name"],
+		serverUpdateScheduleResource["rrule"],
+		serverUpdateScheduleResource["update_name"],
+		retentionPeriod,
+	)
+}
+
+func resourceConfigWithUpdate() string {
+	return fmt.Sprintf(`
+				%s
+
+				resource "stackit_server_update_schedule" "test_schedule" {
+					project_id = "%s"
+					server_id  = "%s"
+					name  = "%s"
+ 				 	rrule = "%s"
+                    enabled = false
+                    update_properties = {
+                        name = "%s"
+                        retention_period = 20 
+                        volume_ids = null
+                    }
+				}
+				`,
+		testutil.ServerUpdateProviderConfig(),
+		serverUpdateScheduleResource["project_id"],
+		serverUpdateScheduleResource["server_id"],
+		serverUpdateScheduleResource["update_schedule_name"],
+		serverUpdateScheduleResource["rrule"],
+		serverUpdateScheduleResource["update_name"],
+	)
+}
```

Comment:
Is this extra function needed? Couldn't we use the one above with extra argument(s) for the fields which are updated?